lib/synapse/config_generator/haproxy.rb

```diff
@@ -961,17 +973,28 @@ def validate_haproxy_stanza(stanza, stanza_type, service_name)
 
     # generates an individual stanza for a particular watcher
     def generate_frontend_stanza(watcher, config)
-      unless watcher.haproxy.has_key?("port")
+      unless watcher.generator_config['haproxy'].has_key?("port")
         log.debug "synapse: not generating frontend stanza for watcher #{watcher.name} because it has no port defined"
         return []
+      else
+        port = watcher.generator_config['haproxy']['port']
```

Comment:
So ... I have plans over in #220 to allow a null port to be semantically different from no port. Basically for unix socket listeners I need to not supply a port, but we still want the frontend and to use bind_address.